wallet/createtx.go

```diff
@@ -446,3 +433,76 @@ func validateMsgTx(tx *wire.MsgTx, prevScripts [][]byte,
 	}
 	return nil
 }
+
+// SelectableCoin is an interface that represents a coin that can be selected
+// for coin selection.
+type SelectableCoin interface {
+	// Output returns the transaction output of the coin.
+	Output() *wire.TxOut
+}
+
+// SelectableCredit is a wrapper around a wtxmgr.Credit that implements the
+// SelectableCoin interface.
+type SelectableCredit struct {
+	*wtxmgr.Credit
+}
+
+// Output returns the transaction output of the credit.
+func (s *SelectableCredit) Output() *wire.TxOut {
+	return &wire.TxOut{
+		Value:    int64(s.Amount),
+		PkScript: s.PkScript,
+	}
+}
+
+// sortByAmount is a generic sortable type for sorting coins by their amount.
+type sortByAmount[T SelectableCoin] []T
+
+func (s sortByAmount[T]) Len() int { return len(s) }
+func (s sortByAmount[T]) Less(i, j int) bool {
+	return s[i].Output().Value < s[j].Output().Value
+}
+func (s sortByAmount[T]) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+
+// ArrangeCoins takes a list of coins and arranges them according to the
+// specified coin selection strategy and fee rate.
+func ArrangeCoins[T SelectableCoin](coinSelectionStrategy CoinSelectionStrategy,
+	eligible []T, feeSatPerKb btcutil.Amount) ([]T, error) {
+
+	switch coinSelectionStrategy {
+	// Pick largest outputs first.
+	case CoinSelectionLargest:
```

Comment:
We could also go one step further here and make `CoinSelectionStrategy` itself an interface. This would allow other custom selection strategies in theory, such as spending the oldest coins first, or selecting random coins, etc.

Reply:
Makes sense, will do.